Trappist/src/Promact.Trappist.Repository/Reports/ReportRepository.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Promact.Trappist.DomainModel.DbContext;
+using Promact.Trappist.DomainModel.Models.TestConduct;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Promact.Trappist.DomainModel.Enum;
+using Promact.Trappist.DomainModel.Models.Test;
+
+namespace Promact.Trappist.Repository.Reports
+{
+    public class ReportRepository : IReportRepository
+    {
+        #region Private Members
+        private readonly TrappistDbContext _dbContext;
+        #endregion
+
+        #region Constructor
+        public ReportRepository(TrappistDbContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+        #endregion
+
+        #region Public Method
+        public async Task<Test> GetTestNameAsync(int id)
+        {
+            return await _dbContext.Test.FindAsync(id);
+
+        }
+
+        public async Task<IEnumerable<TestAttendees>> GetAllTestAttendeesAsync(int id)
+        {
+            return await _dbContext.TestAttendees.Where(t => t.TestId == id).Include(x => x.Report).ToListAsync();
+        }
+
+        public async Task SetStarredCandidateAsync(int id)
+        {
+            var studentToBeStarred = await _dbContext.TestAttendees.FindAsync(id);
+            studentToBeStarred.StarredCandidate = !studentToBeStarred.StarredCandidate;
+            await _dbContext.SaveChangesAsync();
+        }
+
+        public async Task SetAllCandidateStarredAsync(bool status, int selectedTestStatus, string searchString)
+        {
+            List<TestAttendees> attendeeList;
+
+            if (searchString == null)
+                searchString = "";
+            else
+                searchString = searchString.ToLower();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define the locale to be used in this string operation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1449)